agent helper: relax device access for hardware based authentication

Maryse47 opened the pull request:
Various pam modules may use physical hardware like tokens, cameras, fingerprint readers
for authentication thus they need unconstrained /dev access.

They aren't enabled by default in distros pam stack for obvious reasons but their existence
shouldn't be ignored and supported authentication methods limited to plain password.

Fixes https://github.com/polkit-org/polkit/issues/622

Fixes https://github.com/polkit-org/polkit/issues/623

Maryse47 commented:
> Also, the PrivateDevices=no part is not wrong (in contrary to what https://github.com/polkit-org/polkit/pull/626#discussion_r2640471455) in the context of enabling the use of YubiKey

Yes, all parts of this PR are right in this context.

> I would suggest close this PR and open a new PR to the [Yubico/pam-u2f](https://github.com/Yubico/pam-u2f) project that adds a drop-in configuration file that includes a correct set of properties that will allow the access to only the resources required by the pam-u2f process

I would like to see a linux distro accepting such polkit patches by 3rd party package first.

YHNdnzj commented:
Well, I'd argue that this is an inherent flaw in PAM workflow: modules accessing all sorts of resources are loaded into a single process, hence we effectively lose the ability to derive a common sandboxing denominator. This PR, even if merged, only fixes some very specific case and won't scale - it's not inconceivable that a module might need to access network. A way better path forward would be to rewrite modules to operate IPC-based, just as precedented by https://github.com/thkukuk/account-utils. I'm not convinced this is something to be addressed/addressable upstream. As proposed downstream modules can ship drop-ins for a temporary workaround.

Maryse47 commented:
I updated this PR and changed description. Now it only removes `PrivateDevices=yes`, rest of changes are just cleanup related to it. I think supporting any kind of 2FA is important in this day and age and that requires /dev access.

bluca reviewed (changes requested):
> I updated this PR and changed description. Now it only removes `PrivateDevices=yes`, rest of changes are just cleanup related to it. I think supporting any kind of 2FA is important in this day and age and that requires /dev access.

Still wrong. Please drop that change, it belongs in packages that require it, not here.

Maryse47 commented:
Maybe tell that to distro maintainers who don't want this in their packages.

bluca commented on data/polkit-agent-helper@.service.in:
```diff
@@ -14,11 +16,11 @@ LimitMEMLOCK=0
 MemoryDenyWriteExecute=yes
 NoNewPrivileges=yes
-PrivateDevices=yes
+PrivateDevices=no
```
The cleanup above is fine
